traffic_ops/traffic_ops_golang/acme/acme_account.go

```diff
@@ -0,0 +1,261 @@
+	"github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api"
+	"github.com/go-ozzo/ozzo-validation"
+	"net/http"
+)
+
+type AcmeAccount struct {
+	Email      *string `json:"email" db:"email"`
+	PrivateKey *string `json:"private_key" db:"private_key"`
+	Uri        *string `json:"uri" db:"uri"`
+	Provider   *string `json:"provider" db:"provider"`
+}
+
+func (aa *AcmeAccount) Validate(tx *sql.Tx) error {
+
+	errs := validation.Errors{
+		"email":       validation.Validate(aa.Email, validation.Required),
+		"private_key": validation.Validate(aa.PrivateKey, validation.Required),
+		"uri":         validation.Validate(aa.Uri, validation.Required),
+		"provider":    validation.Validate(aa.Provider, validation.Required),
+	}
+
+	return util.JoinErrs(tovalidate.ToErrors(errs))
+}
+
+func (aa *AcmeAccount) ValidateUpdate(tx *sql.Tx) error {
+
+	errs := validation.Errors{
+		"email":    validation.Validate(aa.Email, validation.Required),
+		"provider": validation.Validate(aa.Provider, validation.Required),
+	}
+
+	return util.JoinErrs(tovalidate.ToErrors(errs))
+}
+
+const readQuery = `SELECT email, private_key, uri, provider FROM acme_account`
+const createQuery = `INSERT INTO acme_account (email, private_key, uri, provider) VALUES (:email, :private_key, :uri, :provider) RETURNING email, provider`
+const updateQuery = `UPDATE acme_account SET private_key=:private_key, uri=:uri WHERE email=:email and provider=:provider RETURNING email, provider`
+const deleteQuery = `DELETE FROM acme_account WHERE email=$1 and provider=$2`
+
+func Read(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	var acmeAccounts []AcmeAccount
+	rows, err := tx.Query(readQuery)
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New("querying acme accounts: "+err.Error()))
+		return
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var acct AcmeAccount
+		if err = rows.Scan(&acct.Email, &acct.PrivateKey, &acct.Uri, &acct.Provider); err != nil {
+			api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, errors.New("scanning acme accounts: "+err.Error()))
+			return
+		}
+		acmeAccounts = append(acmeAccounts, acct)
+	}
+
+	api.WriteResp(w, r, acmeAccounts)
+}
+
+func Create(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, inf.Tx.Tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	tx := inf.Tx.Tx
+
+	var acmeAccount AcmeAccount
+	if err := json.NewDecoder(r.Body).Decode(&acmeAccount); err != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, err, nil)
+		return
+	}
+
+	if validErr := acmeAccount.Validate(tx); validErr != nil {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, validErr, nil)
+		return
+	}
```

Comment:
One of the reasons to give something a `.Validate(*sql.Tx) error` method is so that you can use `github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/api.Parse` instead of doing these two blocks yourself.